src/AI/AICombatUtils.java

```diff
@@ -144,40 +145,99 @@ public static int findMaxStrikeWeaponRange(Commander co)
   }
 
   /**
-   * Finds a kill on the designated unit, if available.
-   * @return The lethal combination of units organized by strike location, null on failure.
+   * Finds a kill on the designated unit, if available; considers the cheapest units first.
+   * <p>Does not consider attacks from fogged tiles.
+   * <p>Prunes any unnecessary attacks and empty spaces from the return value.
+   * <p>Does not consider launchable units
+   * @param target Your victim
+   * @param co Whose turn it is now
+   * @param attackCandidates The units CO is willing and able to commit
+   * @param excludedSpaces Any tiles you don't want to consider attacks from
+   * @return The lethal combination of units organized by strike location, or null on failure.
    */
-  public static HashMap<XYCoord, Unit> findAssaultKill(
+  public static HashMap<XYCoord, Unit> findMultiHitKill(
                                    GameMap gameMap, Unit target,
                                    Commander co, Collection<Unit> attackCandidates,
                                    Collection<XYCoord> excludedSpaces)
   {
     if( target.getHP() < 1 ) // Try not to pick fights with zombies
       return null;
 
-    int minRange = 1;
-    ArrayList<XYCoord> coordsToCheck =
+    final int minRange = 1;
+    final XYCoord targetCoord = new XYCoord(target);
+    HashSet<XYCoord> coordsToCheck = new HashSet<XYCoord>(
         Utils.findLocationsInRange(gameMap,
-                                   new XYCoord(target.x, target.y),
-                                   minRange, findMaxStrikeWeaponRange(co));
+                                   targetCoord,
+                                   minRange, findMaxMobileWeaponRange(co))
+        );
 
-    HashMap<XYCoord, Unit> neededAttacks = new HashMap<XYCoord, Unit>();
-    // Figure out where we can attack from, and include attackers already in range by default.
-    for( XYCoord xyc : coordsToCheck )
+    // Consider the cheapest units first.
+    PriorityQueue<Unit> attackers = new PriorityQueue<Unit>(11, new AIUtils.UnitCostComparator(true));
```

Comment:
Seems like you just changed quite a bit more than simply adding docs. Which is always a bit nervous right before merging in a thing.
I'm sure the unit tests for this function still pass though, so I'm sure there's no real cause for alarm.

Reply:
Yeah, I did. Made Wally take under 40 seconds for all the current defaults in Fight Club, too.
And yes, the AI tests all still pass.

Reply:
...I suppose there aren't any real *unit* tests, but it'd probably be more work than it's worth to set up a proper unit test.

Reply:
Cool. And yeah, probably.